src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -270,21 +294,64 @@ def register_path_argument(scope, default_file_param=None, options_list=None):
-register_cli_argument('storage queue policy', 'permission', options_list=('--permissions',), type=validate_queue_permission, help='permissions granted: (r)ead (a)dd (u)pdate (p)rocess [delete]. Can be combined')
 
 register_cli_argument('storage message', 'queue_name', queue_name_type)
 register_cli_argument('storage message', 'message_id', options_list=('--id',))
 register_cli_argument('storage message', 'content', type=unicode_string)
 
-###################################################################################################
-
-# TODO: Address these in Storage PR3 (week of 8/15/16)
-resource_types_type = CliArgumentType(type=validate_resource_types, help='the resource types the SAS is applicable for. Allowed values: (s)ervice (c)ontainer (o)bject. Can be combined.')
-register_cli_argument('storage account generate-sas', 'resource_types', resource_types_type)
-
-services_type = CliArgumentType(type=validate_services, help='the storage services the SAS is applicable for. Allowed values: (b)lob (f)ile (q)ueue (t)able. Can be combined.')
-register_cli_argument('storage account generate-sas', 'services', services_type)
-
-ip_type = CliArgumentType(help='specifies the IP address or range of IP addresses from which to accept requests', type=validate_ip_range)
-register_cli_argument('storage', 'ip', ip_type)
- 
\ No newline at end of file
+for item in ['account', 'blob', 'container', 'file', 'share', 'table', 'queue']:
+    register_cli_argument('storage {} generate-sas'.format(item), 'ip', help='Specifies the IP address or range of IP addresses from which to accept requests. Supports only IPv4 style addresses.', type=ipv4_range_type)
+    register_cli_argument('storage {} generate-sas'.format(item), 'expiry', help='Specifies the UTC datetime (Y-m-d\'T\'H:M\'Z\') at which the SAS becomes invalid. Do not use if a stored access policy is referenced with --id that specifies this value.', type=datetime_string_type)
+    register_cli_argument('storage {} generate-sas'.format(item), 'start', help='Specifies the UTC datetime (Y-m-d\'T\'H:M\'Z\') at which the SAS becomes valid. Do not use if a stored access policy is referenced with --id that specifies this value. Defaults to the time of the request.', type=datetime_string_type)
+    register_cli_argument('storage {} generate-sas'.format(item), 'protocol', options_list=('--https-only',), help='Only permit requests made with the HTTPS protocol. If omitted, requests from both the HTTP and HTTPS protocol are permitted.', action='store_const', const='https')
+
+register_cli_argument('storage container generate-sas', 'id', options_list=('--policy-name',), help='The name of a stored access policy within the container\'s ACL.', completer=get_storage_acl_name_completion_list(BaseBlobService, 'container_name', 'get_container_acl'))
```

Comment:
I'm going to refactor this code a bit, but I don't think we should move the registration to 'storage' in order to prevent inadvertent aliasing. If another command was released that did use id as a policy name that was under storage, then I could see it. We've had too many aliasing snafus in the past IMO. 

---
In reply to: [75403406](https://github.com/Azure/azure-cli/pull/702#discussion_r75403406) [](ancestors = 75403406)